edx_rest_framework_extensions/authentication.py

```diff
@@ -0,0 +1,87 @@
+""" Authentication classes. """
+
+import logging
+
+import requests
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from rest_framework import exceptions
+from rest_framework.authentication import get_authorization_header, BaseAuthentication
+from rest_framework.status import HTTP_200_OK
+
+logger = logging.getLogger(__name__)
+User = get_user_model()
+
+
+class BearerAuthentication(BaseAuthentication):
+    """
+    Simple token based authentication.
+
+    This authentication class is useful for authenticating an OAuth2 access token
+     against a remote authentication provider.
+
+    Clients should authenticate by passing the token key in the "Authorization"
+    HTTP header, prepended with the string "Bearer ".
+
+    Examples:
+        Authorization: Bearer 401f7ac837da42b97f613d789819ff93537bee6a
+    """
+
+    def get_provider_url(self):
+        """ Returns the root URL of the OAuth2 provider. """
+        return getattr(settings, 'OAUTH2_PROVIDER_URL', None)
+
+    def authenticate(self, request):
+        provider_url = self.get_provider_url()
+        if not provider_url:
+            return None
+
+        provider_url = provider_url.strip('/')
```

Comment:
Stripping off the leading slashes from the provider URL introduces possibly unexpected behavior.  A site-relative absolute URL like `/path/to/endpoint` will be changed to the relative URL `path/to/url`, which has a very different meaning.  A protocol-relative URL like `//facebook.com/path/to/endpoint/` will be changed to the very different url `facebook.com/path/to/endpoint`.  Unless this is guarded against elsewhere, and specifically desired behavior for reasons I'm not seeing, I'd suggest using `provider_url.rstrip()` to only trim trailing slashes.